### PR TITLE
refactor(cli): use Cobra framework for CLI commands

### PR DESCRIPTION
**Pull request by educhastenier**
Cobra is the de-facto framework for writing Command-based Go programs.
It is widely used by big projects (Github Cli, Docker client...)

The main idea is to clearly separate command code into their own code file
and to have the syntax, usage and help, validation... automatically generated for us.

Reference https://github.com/spf13/cobra

⚠️ The main executable command is not named `bonita-packager` anymore but `bonita` and has a sub-command named `package`
which lead to its main usage being `bonita package ...`

### CODE REVIEW

File: cmd/version.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+var (
+	version    string
+	versionCmd = &cobra.Command{
+		Use:   "version",
+		Short: "Print bonita cli version",
+		Long:  `Print the version of the Bonita CLI command`,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("Bonita CLI", Version)
```
Comment (educhastenier) [inline cmd/version.go:20]: Can we have the same output with the `--version` syntax?

File: README.md
Comment (educhastenier) [inline README.md:1]: Is there already some documentation that would need to be updated with this new syntax?

File: cmd/deploy.go
Comment (akantcheff) [inline cmd/deploy.go:1]: I don't think we need this file. If it is not used it is dead code :p

Comment (educhastenier) [reply cmd/deploy.go:1]: removed

File: cmd/root.go [hunk 2]
```diff
@@ -41,4 +41,5 @@ func init() {
 
 	RootCmd.AddCommand(pkg.PackageCmd)
 	RootCmd.CompletionOptions.DisableDefaultCmd = true
+	RootCmd.SetVersionTemplate(fmt.Sprintf("Bonita CLI %s\n", Version))
```
Comment (educhastenier) [inline cmd/root.go:44]: To customize the `--version` output

**Review by akantcheff — APPROVED** [review]
LGTM

**Pull request merged**